src/hotspot/cpu/aarch64/macroAssembler_aarch64.hpp

```diff
@@ -1084,6 +1084,34 @@ class MacroAssembler: public Assembler {
 public:
   // Calls
 
+  // Trampoline_call generates a call of the provided address.
+  // The address must be inside the code cache.
+  // Supported entry.rspec():
+  // - relocInfo::runtime_call_type
+  // - relocInfo::opt_virtual_call_type
+  // - relocInfo::static_call_type
+  // - relocInfo::virtual_call_type
+  //
+  // If the distance to the address can exceed the branch range
+  // (128M for the release build, 2M for the debug build; see branch_range definition)
+  // for direct calls(BL), a special code with BR (trampoline) is put in the stub code section.
+  // The call is redirected to it. When CodeBuffer is copied to CodeCache, the distance to
+  // callee's address is checked to bypass the trampoline by replacing the call of the trampoline
+  // with the call of the target.
+  //
+  // Trampoline_call is most suitable for calls of Java methods. Java calls callees can be changed
+  // to the interpreter or different versions of a compiled method. Those callees can be
```

Comment:
This is not true. Trampoline calls are needed for calls to the runtime too.

Reply:
Why? What are the cases? This is not clear from sources.

Reply:
> This commentary is garbled, misleading, and very confusing.

Could you please be more neutral in your words?

> I suspect that anyone confused about how trampoline calls work will be even more confused after trying to read this. 

This is why we have the review process to make things better.

> Can you please explain what aspects of trampolines you're trying to clarify? Let's have a conversation about how to document this.

I am trying to clarify why we have two ways for long calls, pros/cons, which of two can be optimized. This mechanism of long call is very important part but it is not documented.